test/main/chains/gas/index.test.js

```diff
@@ -0,0 +1,227 @@
+      // index in array represents distance away from current block
+      feeHistory = [
+        { baseFee: 8, gasUsedRatio: 0.073, rewards: [2587202560] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, gasUsedRatio: 1.122, rewards: [2000000000] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, rewards: [] },
+        { baseFee: 8, gasUsedRatio: 1.2239, rewards: [1000000000] },
+        { baseFee: 182, rewards: [] }
+      ]
+
+      const rewards = feeHistory.reduce(
+        (acc, { rewards }) => (rewards.length ? acc.concat(intToHex(rewards[0])) : acc),
+        []
+      )
+
+      expect(rewards.includes(await calculateMaxPriority())).toBe(true)
+    })
+
+    it('uses the priority fee from the latest block when no eligible blocks are available', async () => {
+      feeHistory = [
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 182, rewards: [] }
+      ]
+
+      const { maxBaseFeePerGas, maxPriorityFeePerGas } = await calculateGas()
+      expect(maxBaseFeePerGas).toBe('0xe7')
+      expect(maxPriorityFeePerGas).toBe('0x0')
+    })
+
+    describe('polygon gas calculator', () => {
+      beforeEach(() => {
+        gasCalculator = createGasCalculator(137)
+      })
+      it('should update polygon maxPriorityFee below 30 gwei to 30 gwei', async () => {
+        feeHistory = [
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+          { baseFee: 8, gasUsedRatio: 0.07942918604838942, rewards: [1000000000] },
+          { baseFee: 8, gasUsedRatio: 0.990240614960509, rewards: [1000000000] },
+          { baseFee: 182, rewards: [] }
+        ]
+        expect(await calculateMaxPriority()).toBe(gweiToHex(30))
+      })
+      it('does not change any maxPriorityFeePerGas that is over 30 gwei', async () => {
+        feeHistory = [
+          { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
```

Comment:
is it possible to use a simpler set of test data to make it more illustrative what is happening here? if not, no worries just might make the test a little more readable in the future